Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -388,6 +388,17 @@ theorem finPiFinEquiv_single {m : ℕ} {n : Fin m → ℕ} [∀ i, NeZero (n i)]
   rintro x hx
   rw [Pi.single_eq_of_ne hx, Fin.val_zero', zero_mul]
 
+/-- Equivalence between the Sigma type `(i : Fin m) × Fin (n i)` and `Fin (∑ i : Fin m, n i)`. -/
+def finSigmaFinEquiv {m : ℕ} {n : Fin m → ℕ} : (i : Fin m) × Fin (n i) ≃ Fin (∑ i : Fin m, n i) :=
+  match m with
+  | 0 => @Equiv.equivOfIsEmpty _ _ _ (by simp; exact Fin.isEmpty')
+  | Nat.succ m =>
+    calc _ ≃ _ := (@finSumFinEquiv m 1).sigmaCongrLeft.symm
```

Comment:
If you don't write any of the expressions, `calc` block is a bit superfluous; I'd rather just compose the parts from `:=` on the right.

Reply:
Sure, you think

```
      (@finSumFinEquiv m 1).sigmaCongrLeft.symm.trans <|
      (Equiv.sumSigmaDistrib _).trans <|
      (finSigmaFinEquiv.sumCongr (Equiv.uniqueSigma _)).trans <|
      finSumFinEquiv.trans <|
      finCongr (Fin.sum_univ_castSucc n).symm
```

is best then?